Add support in suffix / prefix per processed file

dudabone opened the pull request:
Currently suffix / prefix options are passed as strings and are being applied to every processed file. This change allows passing an these options as an object, where the key is the file to which this option applies to and the value is the prefix/suffix to add. e.g:

``` javascript
suffix: {
  '../path/to/file1': 'obj1 = getMessages();',
  '../path/to/file2': 'obj2 = getMessages();'
}
```

dudabone committed:
Add support in suffix / prefix per processed file

Currently suffix / prefix options are passed as strings and are being applied to every processed file. This change allows passing an these options as an object, where the key is the file to which this option applies to and the value is the prefix/suffix to add. e.g:

suffix: {
  '../path/to/file1': 'obj1 = getMessages();',
  '../path/to/file2': 'obj2 = getMessages();'
}

MadaraUchiha commented:
Thanks for the PR!

This requires the consumer specifying a prefix for each and every `.properties` file in their projects. Those who don't will get `"undefined"` (coerced into a string) as their prefix/suffix, which is a bug.

A potentially better solution is to use some sort of pattern (either glob or regex) as the key, so that you have something like this:

MadaraUchiha commented:
Thanks for the PR!

This requires the consumer specifying a prefix for each and every `.properties` file in their projects. Those who don't will get `"undefined"` (coerced into a string) as their prefix/suffix, which is a bug.

---

A potentially better solution is to use some sort of pattern (either glob or regex) as the key, so that you have something like this:

```
suffix: {
  'messages*.properties': 'obj1 = getMessages();',
  '*.properties': 'obj2 = getMessages();'
}
```

dudabone commented:
Sounds right, especially considering the fact that my code doesn't work currently with relative paths, so it is practically quite useless now :( 

If we do go with glob / regex - I wonder what should be the behavior in case a file matches more than one pattern.

dudabone committed:
Changed prefix/suffix options to support globs

Allows passing prefix/suffix options as an object. The key is a glob pattern and the value is the suffix / prefix. If a processed file matches one of the glob patterns, its value will be used as a suffix/prefix. First match wins.

dudabone committed:
Added "minimatch" as a dependency
